src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -23,23 +30,46 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
     public DeleteCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME);
 
-        if (argMultimap.getValue(PREFIX_NAME).isEmpty()) {
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            String preamble = argMultimap.getPreamble().trim();
+            if (!preamble.isEmpty()) {
+                throw new ParseException(String.format(MESSAGE_MULTIPLE_WAYS_FORBIDDEN,
+                        DeleteCommand.COMMAND_WORD));
+            }
+            String str = argMultimap.getValue(PREFIX_NAME).get();
+
             try {
-                Index index = ParserUtil.parseIndex(args);
-                return new DeleteCommand(index);
-            } catch (ParseException ex) {
-                throw new ParseException(
-                        String.format(MESSAGE_NAME_FIELD_MISSING, DeleteCommand.MESSAGE_USAGE), ex);
+                Name name = ParserUtil.parseName(str);
+                return new DeleteCommand(name);
+            } catch (Exception ex) {
+                throw ex; // why it became successfull
+                // did not use the 2 parameter exception
```

Comment:
This one is delete n/James&
Use of either index of name. But fails because Invalid name

Maybe a more descriptive message